### ISSUE TITLE
TSQL: Add support for column alias with equals (`=`) operator to rule L013

### ISSUE THREAD

**Issue by infused-kim** [sqlfluff/sqlfluff#2552]
In addition to the ansi alias syntax (`getdate() as ansi_alias`), TSQL also supports aliases using the equals (`=`) operator: `equal_alias = getdate()`.

I propose to add support for that alias syntax and perhaps add a rule that let's users select a preferred style.

I would be happy to try implementing this if someone could give me a pointer into the direction of how this could be accomplished. 

## Steps to Reproduce
- Set dialect to tsql
- Lint the code below

```sql
select
    equals_alias = getdate(),
    getdate() as ansi_alias
from my_table
```

## Expected Behaviour
To not trigger `L013`.

## Observed Behaviour
Rule `L013` is triggered:
```log
L:   2 | P:   5 | L013 | Column expression without alias. Use explicit `AS`
                       | clause.
```

## Dialect
TSQL

## Version

- sqlfluff, version 0.9.4
- Python 3.9.9
- dbt 1.0.0

## Configuration

```ini
[sqlfluff]
# Run `sqlfluff dialects` to get a list of all dialects
dialect = tsql

# Use dbt as the templater
templater = dbt

# Use jinja templater instead of dbt
#templater = jinja

[sqlfluff:templater:jinja]
# Make dbt default macros like available like `ref`
apply_dbt_builtins = true

# Load custom macros from the macro directory
load_macros_from_path = ./macros/

# Allow mocking dbt_utils and other packages through
# python files placed in this dir
library_path = ./sqlfluff_libs/

# Mock variables that are used in dbt
[sqlfluff:templater:jinja:context]
target = prod
```

**Comment by tunetheweb** [comment]
Related #2487

**Comment by tunetheweb** [comment]
Something like this seems to work:

```py
@tsql_dialect.segment()
class AltAliasExpressionSegment(BaseSegment):
    """An alternative alias clause as used by tsql"""

    type = "alias_expression"
    match_grammar = Sequence(
        OneOf(
            Ref("SingleIdentifierGrammar"),
            Ref("SingleQuotedIdentifierSegment"),
        ),
        Ref("RawEqualsSegment"),
    )


@tsql_dialect.segment(replace=True)
class SelectClauseElementSegment(BaseSegment):
    """An element in the targets of a select statement.

    Overriding ANSI to remove GreedyUntil logic which assumes statements have been
    delimited, and adding alt alias syntax
    """

    type = "select_clause_element"
    # Important to split elements before parsing, otherwise debugging is really hard.
    match_grammar = OneOf(
        # *, blah.*, blah.blah.*, etc.
        Ref("WildcardExpressionSegment"),
        OneOf(
            Sequence(
                Ref("AltAliasExpressionSegment"),
                Ref("BaseExpressionElementGrammar"),
            ),
            Sequence(
                Ref("BaseExpressionElementGrammar"),
                Ref("AliasExpressionSegment", optional=True),
            ),
        ),
    )

    get_alias = ansi_dialect.get_segment("SelectClauseElementSegment").get_alias
```

**Comment by fdw** [comment]
Awesome, thank you! 🙂
